File: Othello-Src-Python/display.py
#!/usr/bin/env python3

import board as board

# Color Definition
B_GREEN_F_BLACK = "\033[0;30;42m"
B_BLACK = "\033[0;40m"
B_WHITE = "\033[0;47m"
NORMAL = "\033[0m"
CLEAR = "\033[2J\033[1;1H"


def clear():
    """ Clear the Screen """
    print(CLEAR, end='')


def display(_board: board.Board):
    """ Print the Board """
    if(type(_board) == board.Board):
        print('  ', end='')
        for i in range(board.BOARD_W):
            print("%d " % (i), end='')
        print('')

        for r in range(board.BOARD_H):
            print(NORMAL+" ", end='')
            for _ in range(board.BOARD_W*2+1):
                print(B_GREEN_F_BLACK+"=", end='')
            print(NORMAL)

            print(NORMAL+"%d" % (r), end='')

            for c in range(board.BOARD_W):
                status = _board.get(r, c)
                if(status == board.Status.ILLEGAL):
                    print(B_GREEN_F_BLACK+"|X", end='')
                elif(status == board.Status.BLACK):
                    print(B_GREEN_F_BLACK+"|"+B_BLACK+" ", end='')
                elif(status == board.Status.WHITE):
                    print(B_GREEN_F_BLACK+"|"+B_WHITE+" ", end='')
                else:
                    print(B_GREEN_F_BLACK+"| ", end='')

            print(B_GREEN_F_BLACK+"|")

        print(NORMAL+" ", end='')
        for _ in range(board.BOARD_W*2+1):
            print(B_GREEN_F_BLACK+"=", end='')
        print(NORMAL)

File: Othello-Src-Cpp/src/display.cpp
#include <board.h>
#include <display.h>
#include <cstdio>

/*
    Clear the Screen
*/
void Display::clear() { printf(CLEAR); }

/*
    Print the board
*/
template <>
void Display::print(Board &_board) {
  printf("  ");
  for (int i=0; i< BOARD_W; i++) printf("%d ", i);
  printf("\n");

  for (int i = 0; i < BOARD_H; i++) {
    printf(NORMAL " ");
    for (int k = 0; k < BOARD_W * 2 + 1; k++) printf(B_GREEN_F_BLACK "=");
    printf("\n" NORMAL);

    printf(NORMAL "%d", i);

    for (int j = 0; j < BOARD_W; j++) {
      Status status = _board.get(i, j);
      switch (status) {
        case Status::ILLEGAL:
          printf(B_GREEN_F_BLACK "|X");
          break;
        case Status::BLACK:
          printf(B_GREEN_F_BLACK "|" B_BLACK " ");
          break;
        case Status::WHITE:
          printf(B_GREEN_F_BLACK "|" B_WHITE " ");
          break;
        default:
          printf(B_GREEN_F_BLACK "| ");
          break;
      }
    }

    printf(B_GREEN_F_BLACK "|\n");
  }

  printf(NORMAL " ");
  for (int k = 0; k < BOARD_W * 2 + 1; k++) printf(B_GREEN_F_BLACK "=");
  printf("\n" NORMAL);
}